pkg/models/event.go

```diff
@@ -13,7 +13,9 @@ import (
 // Event is a wrapper of contract.Event to provide more Binary related operation in Device Service.
 type Event struct {
 	contract.Event
-	EncodedEvent []byte
+	EncodedEvent  []byte
+	CorrelationId string
+	Checksum      string
```

Comment:
Do we need these fields?

Reply:
The field `Checksum` is used thrice in the following func, which we can revisit in light of the point regarding using a general `Reader`, as needed:
```
func (cr cborReader) Read(reader io.Reader, ctx *context.Context) (dsModels.Event, error) {
```
But yes, the field `CorrelationId` is _not_ used. Removed it.